schema/test/unit/tables/project_revision_test.sql

```diff
@@ -46,6 +49,100 @@ select throws_ok(
   'prevents delete if the change status is committed'
 );
 
+-- test project_revision_001_add_revision_type
+truncate table cif.project_revision cascade;
+
+-- use the contents of the revert file to set the database to a state where revision_type isn't required
+alter table cif.project_revision drop column revision_type, drop column comments;
+
+select columns_are(
+  'cif',
+  'project_revision',
+  ARRAY[
+    'id',
+    'project_id',
+    'change_status',
+    'change_reason',
+    'is_first_revision',
+    'created_at',
+    'created_by',
+    'updated_at',
+    'updated_by'
+  ],
+  'revision_type and comments column were reverted and do not exist in cif.project_revision'
+);
+
+insert into cif.project(id, project_name, operator_id, funding_stream_rfp_id, project_status_id, proposal_reference, summary)
+overriding system value
+  values (
+    3,'test-project-333',
+    (select id from cif.operator limit 1),
+    (select id from cif.funding_stream_rfp limit 1),
+    (select id from cif.project_status limit 1),
+    'rfp333',
+    'summary333'
+  ),
+  (
+    4,'test-project-444',
+    (select id from cif.operator limit 1),
+    (select id from cif.funding_stream_rfp limit 1),
+    (select id from cif.project_status limit 1),
+    'rfp444',
+    'summary444'
+  );
+
+insert into cif.project_revision(project_id)
+values
+  (3),
+  (4);
+
+-- deploy the project_revision_001_add_revision_type migration
+alter table cif.project_revision disable trigger _100_committed_changes_are_immutable, disable trigger _100_timestamps;
+
+alter table cif.project_revision
+add column revision_type varchar(1000) not null references cif.revision_type(type) default 'General Revision',
+add column comments varchar(10000);
+
+alter table cif.project_revision enable trigger _100_committed_changes_are_immutable, enable trigger _100_timestamps;
+
+
+select columns_are(
+  'cif',
+  'project_revision',
+  ARRAY[
+    'id',
+    'project_id',
+    'change_status',
+    'change_reason',
+    'is_first_revision',
+    'comments',
+    'created_at',
+    'created_by',
+    'updated_at',
+    'updated_by',
+    'revision_type'
+  ],
+  'columns in cif.project_revision match expected columns after migration project_revision_001_add_revision_type'
+);
+
+
+select is(
+  (select count(*) from cif.project_revision where revision_type='General Revision'),
+  2::bigint,
+  'project_revision_001_add_revision_type adds the General Revision default revision_type to revisions that do not have a type');
+
+
+truncate table cif.project_revision cascade;
+insert into cif.project_revision(project_id, revision_type)
+values
+  (3, 'Amendment'),
+  (4, 'Minor Revision');
+
+select is(
+  (select count(*) from cif.project_revision where revision_type='General Revision'),
+  0::bigint,
+  'project_revision_001_add_revision_type does not change the revision_type of projects that already have revision types');
+
```

Comment:
I'm a little confused about what we test here, the migration adds the `revision_type` column, so making sure that existing records with a revision_type don't get impacted has little sense. We can probably remove this test